python/cm/adcm_schema.yaml

```diff
@@ -0,0 +1,617 @@
+#                             O B J E C T S
+###############################################################################
+# There are a number of object types: cluster, provider, host and service
+# All of them has the same struture but with small difference
+
+object:
+  match: dict_key_selection
+  selector: type
+  variants:
+    cluster: cluster_object
+    service: service_object
+    host: host_object
+    provider: provider_object
+    adcm: adcm_object
+
+common_object: &common_object
+  match: dict
+  items: &common_object_items
+    type: string
+    name: string
+    version: version_rule
+    display_name: string
+    description: string
+    edition: string
+    license: string
+    adcm_min_version: version_rule
+    config: config_obj
+    actions: actions_dict
+  required_items: &common_object_required_items
+    - type
+    - name
+    - version
+
+service_object:
+  <<: *common_object
+  items:
+    <<: *common_object_items
+    import: import_dict
+    export: export
+    components: components_dict
+    required: boolean
+    monitoring: monitoring
+
+cluster_object:
+  <<: *common_object
+  items:
+    <<: *common_object_items
+    upgrade: upgrade_list
+    import: import_dict
+    export: export
+
+host_object:
+  <<: *common_object
+  items:
+    <<: *common_object_items
+
+provider_object:
+  <<: *common_object
+  items:
+    <<: *common_object_items
+    upgrade: upgrade_list
+
+adcm_object:
+  <<: *common_object
+  items:
+    <<: *common_object_items
+    upgrade: upgrade_list
+
+export:
+  match: one_of
+  variants:
+    - string
+    - list_of_string
+
+monitoring:
+  match: set
+  variants:
+    - active
+    - passive
+
+components_dict:
+  match: dict
+  default_item: component_item
+
+component_item:
+  match: one_of
+  variants:
+    - none
+    - component_dict
+
+component_dict:
+  match: dict
+  items:
+    display_name: string
+    monitoring: string
+    constraint: constraint_list
+    params: json
+    requires: comp_req_list
+
+comp_req_list:
+  match: list
+  item: comp_req_item
+
+comp_req_item:
+  match: dict
+  items:
+    service: string
+    component: string
+  required:
+    - component
+
+constraint_list:
+  match: list
+  item: constraint_list_item
+
+constraint_list_item:
+  match: one_of
+  variants:
+    - integer
+    - string
+
+version_rule:
+  match: one_of
+  variants:
+    - integer
+    - string
+    - float
+
+# Upgrade Block
+upgrade_list:
+  match: list
+  item: upgrade_obj
+
+upgrade_obj:
+  match: dict
+  items:
+    name: string
+    description: string
+    versions: version_dict
+    states: states_dict
+    from_edition: any_or_list
+  required_items:
+    - name
+    - versions
+
+version_dict:
+  match: dict
+  items:
+    min: version_rule
+    max: version_rule
+    min_strict: version_rule
+    max_strict: version_rule
+
+states_dict:
+  match: dict
+  items:
+    available: available_state
+    on_success: string
```

Comment:
it's right? now there are tests for different types of `on_success` property

Reply:
yep, that's right